core/api/service/api_service.cpp

```diff
@@ -190,12 +236,25 @@ namespace kagome::api {
 
               // process new request
               self->server_->processData(
-                  str_request,
-                  [session = std::move(session)](
-                      const std::string &response) mutable {
+                  str_request, [&](const std::string &response) mutable {
                     // process response
                     session->respond(response);
                   });
+
+              try {
+                self->for_session(
+                    session->id(),
+                    [&](SessionExecutionContext &session_context) {
+                      if (session_context.messages)
+                        for (auto &msg : *session_context.messages) {
+                          BOOST_ASSERT(msg);
+                          session->respond(*msg);
+                        }
+
+                      session_context.messages.reset();
+                    });
+              } catch (jsonrpc::InternalErrorFault &) {
```

Comment:
Is it okay not to handle exception here? Maybe add some log at least?

Reply:
No, it's ok. The request can be from HTTP session and this session will not be stored with context.